### PR TITLE
test.py: manual cluster pool handling for Python suite

### PR DESCRIPTION
**Pull request by alecco**
From reviews of https://github.com/scylladb/scylladb/pull/12569, avoid using `async with` and access the `Pool` of clusters with `get()`/`put()`.

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -842,33 +842,36 @@ async def run(self, options: argparse.Namespace) -> Test:
...
+                await cluster.release_ips()
+            elif self.is_after_test_ok is False:
+                print("Test {} post-check failed: {}".format(self.name, str(e)))
+                print("Server log of the first server:\n{}".format(self.server_log))
+                logger.info(f"Discarding cluster after failed test %s...", self.name)
+                await self.suite.clusters.steal()
+                await cluster.stop()
+                await cluster.release_ips()
+            else:
+                await self.suite.clusters.put(cluster)
```
Comment (michoecho) [inline test.py:874]: After the patch you are only calling `put` on exceptions. The happy path is leaking clusters.

**Review by michoecho — CHANGES_REQUESTED** [review]

File: test.py (hunk 1 above)
Comment (alecco) [reply test.py:874]: Wrong indentation. Good catch.

**Comment by alecco** [comment]
Fixes in https://github.com/scylladb/scylladb/commit/fa98809ab458e6cf201eb4e4f264658d48f48e1f:

- Fix indentation of non-failure path (try/except/else not inside except)

File: test.py [hunk 2]
```diff
@@ -842,33 +842,36 @@ async def run(self, options: argparse.Namespace) -> Test:
...
+                await cluster.release_ips()
+            elif self.is_after_test_ok is False:
+                print("Test {} post-check failed: {}".format(self.name, str(e)))
+                print("Server log of the first server:\n{}".format(self.server_log))
+                logger.info(f"Discarding cluster after failed test %s...", self.name)
+                await self.suite.clusters.steal()
+                await cluster.stop()
+                await cluster.release_ips()
+        else:
+            await self.suite.clusters.put(cluster)
```
Comment (michoecho) [inline test.py:874]: By changing the indentation you moved the leak from one path to another. Now the happy path isn't leaking, but the path you just removed the `put` from (exception raised, but cluster checks OK) is leaking.

**Review by michoecho — CHANGES_REQUESTED** [review]

File: test.py (hunk 2 above)
Comment (alecco) [reply test.py:874]: In theory either of the conditions should've failed as they are both set to `False` before a test starts. But yes, it makes sense to just stop the cluster for all exceptions. And code is more compact.

Comment (michoecho) [reply test.py:871]: Oh, okay. I thought that it's possible for a test to fail without messing up the cluster, and then the cluster can be reused even after a failure. That's how the code before the patch behaves.

**Comment by alecco** [comment]
Fixes in https://github.com/scylladb/scylladb/commit/f236d518c6f676a4a422a7f92c15901c4c6402d0:

- Added missing argument logger to `clusters.get()`

**Review by kbr-scylla — APPROVED** [review]

**Pull request closed without merging**